lib/taskana-core-test/src/test/java/acceptance/task/create/CreateTaskWithAttachmentAccTest.java

```diff
@@ -0,0 +1,201 @@
+    Attachment attachment = taskService.newAttachment();
+    attachment.setObjectReference(defaultObjectReference);
+    ClassificationSummary classification =
+        classificationService
+            .newClassification(
+                defaultClassificationSummary.getKey(),
+                defaultClassificationSummary.getDomain(),
+                defaultClassificationSummary.getType())
+            .asSummary();
+    attachment.setClassificationSummary(classification);
+    Task taskToCreate = taskService.newTask(defaultWorkbasketSummary.getId());
+    taskToCreate.setClassificationKey(defaultClassificationSummary.getKey());
+    taskToCreate.setPrimaryObjRef(defaultObjectReference);
+    taskToCreate.addAttachment(attachment);
+    taskToCreate = taskService.createTask(taskToCreate);
+
+    classification = taskToCreate.getAttachments().get(0).getClassificationSummary();
+    assertThat(classification).isEqualTo(defaultClassificationSummary);
+  }
+
+  @WithAccessId(user = "user-1-1")
+  @TestFactory
+  Stream<DynamicTest>
+      should_ThrowException_When_CreatingTaskWithInvalidObjectReferenceOfAttachment() {
+    ObjectReference objRefTypeNull =
+        taskService.newObjectReference("Company", "System", "Instance", null, "Value");
+    ObjectReference objRefValueNull =
+        taskService.newObjectReference("Company", "System", "Instance", "Type", null);
+    ObjectReference objRefCompanyNull =
+        taskService.newObjectReference(null, "System", "Instance", "Type", "Value");
+    List<Pair<String, ObjectReference>> valuesForTests =
+        List.of(
+            Pair.of("ObjRef is null", null),
+            Pair.of("Type of objRef is null", objRefTypeNull),
+            Pair.of("Value of objRef is null", objRefValueNull),
+            Pair.of("Company of objRef is null", objRefCompanyNull));
+
+    ThrowingConsumer<Pair<String, ObjectReference>> test =
+        p -> {
+          Task task = taskService.newTask(defaultWorkbasketSummary.getId());
+          task.setClassificationKey(defaultClassificationSummary.getKey());
+          task.setPrimaryObjRef(defaultObjectReference);
+          Attachment attachment = taskService.newAttachment();
+          attachment.setClassificationSummary(defaultClassificationSummary);
+          attachment.setObjectReference(p.getRight());
+          task.addAttachment(attachment);
+          assertThatThrownBy(() -> taskService.createTask(task))
+              .isInstanceOf(InvalidArgumentException.class);
+        };
+
+    return DynamicTest.stream(valuesForTests.iterator(), Pair::getLeft, test);
+  }
+
+  @WithAccessId(user = "user-1-1")
+  @TestFactory
+  Stream<DynamicTest>
+      should_ThrowException_When_CreatingTaskWithInvalidClassificationOfAttachment() {
+    Classification nonExistingClassification =
+        classificationService.newClassification(
+            "123key345", defaultWorkbasketSummary.getDomain(), "non existing type");
```

Comment:
I prefer a more obvious key like `NOT_EXISTING`

Reply:
you can hardcode `DOMAIN_A` since they exist in the configuration

Reply:
please use a proper type to clarify that we're only testing the key here.

Reply:
Ok :)